src/useMemoryRouter.tsx

```diff
@@ -34,23 +36,29 @@ export const useMemoryRouter = (singletonRouter: MemoryRouter, eventHandlers?: M
   // Subscribe to any eventHandlers:
   useEffect(() => {
     if (!eventHandlers) return;
-    const { onRouteChangeStart, onRouteChangeComplete, onPush, onReplace } = eventHandlers;
-    if (!(onRouteChangeStart || onRouteChangeComplete || onPush || onReplace)) return;
+    const { onRouteChangeStart, onRouteChangeComplete, onHashChangeComplete, onHashChangeStart, onPush, onReplace } = eventHandlers;
+    if (!(onRouteChangeStart || onRouteChangeComplete  || onHashChangeStart ||onHashChangeComplete || onPush || onReplace)) return;
 
     if (onRouteChangeStart) singletonRouter.events.on("routeChangeStart", onRouteChangeStart);
     if (onRouteChangeComplete) singletonRouter.events.on("routeChangeComplete", onRouteChangeComplete);
+    if (onHashChangeStart) singletonRouter.events.on("hashChangeStart", onHashChangeStart);
+    if (onRouteChangeComplete) singletonRouter.events.on("hashChangeComplete", onRouteChangeComplete);
```

Comment:
This looks like a typo
```suggestion
    if (onHashChangeComplete) singletonRouter.events.on("hashChangeComplete", onHashChangeComplete);
```

Reply:
oh 😱 - nice find